src/types/timezone.jl

```diff
@@ -67,32 +83,56 @@ TimeZone(::AbstractString, ::Class)
 function TimeZone(str::AbstractString, mask::Class=Class(:DEFAULT))
     # Note: If the class `mask` does not match the time zone we'll still load the
     # information into the cache to ensure the result is consistent.
-    tz, class = get!(_tz_cache(), str) do
-        tz_path = joinpath(_COMPILED_DIR[], split(str, "/")...)
+    ftz, class = get(_ftz_cache(), str, (nothing, Class(:NONE)))
+    if ftz !== nothing
+        _check_class(mask, class)
+        return ftz::FixedTimeZone
+    end
```

Comment:
🤔 hrrmm the downside of the https://github.com/JuliaConcurrent/MultiThreadedCaches.jl approach is that it currently _only_ exposes `get!()` as its interface, for reasons that i cannot fully remember..

So the trick you're doing here wouldn't work. Instead, you'd maybe need to figure out which type of TimeZone it is ahead of time? I don't know if that's possible though, just from its name? I guess you could have a third cache that just caches the type of the timezone?

So it'd be something like:
```julia
    type = get!(_tz_type_cache, str) do
        compute_tz_type(str)
    end
    if type === :FixedTimeZone
        tz, class = get!(ftz_cache, str) do
            compute_fixed_timezone(str)
        end
    elseif type === :VariableTimeZone
        tz, class = get!(vtz_cache, str) do
            compute_variable_timezone(str)
        end
    end
```
Something like that? I'm not sure if splitting things up this way is feasible though?

Reply:
> figure out which type of TimeZone it is ahead of time?

I think that's what's not easy, because we actually create the TimeZones by reading from a timezone-specific file
https://github.com/JuliaTime/TimeZones.jl/blob/f8f39b3100334a0079f525304f7dddc7656204da/src/tzjfile/read.jl#L52-L78

i.e. we have to read the file to know what type it will be, and we don't want to have to read multiple times

so i think we'd have change that `read` function to return the type, then put that value in a third cache, while holding on to the `tz, class`

something like
```julia
    local tz, class
    type = get!(_tz_type_cache, str) do
        tz, class, type = compute_tz(str)
        return type
    end
    if type === :FixedTimeZone
        tz, class = get!(ftz_cache, str) do
            (tz, class)
        end
    elseif type === :VariableTimeZone
        tz, class = get!(vtz_cache, str) do
            (tz, class)
       end
    end
```

Reply:
hmmmm, this way's not looking nice to me 🤔 

how much benefit do we think MultiThreadedCaches.jl is going to add for us?

Reply:
we could maybe go add `get` and `empty!` to the MultiThreadedCaches.jl API to reduce the necessary changes